perun-core/src/test/java/cz/metacentrum/perun/core/bl/ServicesManagerBlImplTest.java

```diff
@@ -117,9 +163,58 @@ public void setUp() throws ServiceExistsException, PrivilegeException {
 		jdbcTemplate.update("insert into facilities(id, name) values (?,?)", testFacilityId2, "Cluster_" + testFacilityId2);
 		facility2.setId(testFacilityId2);
 
-
+		// vo
+		vo = new Vo(0, "ServicesManagerTestVo", "RMTestVo");
+		vo = ((PerunBl)perun).getVosManagerBl().createVo(perunSession, vo);
+		
+		// resource
+		resource = new Resource();
+		resource.setName("ServicesManagerTestResource");
+		resource.setDescription("Testovaci");
+		resource = ((PerunBl)perun).getResourcesManagerBl().createResource(perunSession, resource, vo, facility1);
+		
+		// resource services
+		((PerunBl)perun).getResourcesManagerBl().assignService(perunSession, resource, testService1);
+		
+		// facility_service_destinations
+		destination = ((PerunBl)perun).getServicesManagerBl().getDestinationById(perunSession, testDestinationId1);
+		((PerunBl)perun).getServicesManagerBl().addDestination(perunSession, testService1, facility1, destination);
+		
+		// service package
+		servicesPackage = new ServicesPackage();
+		servicesPackage.setName("ResourcesManagertTestSP");
+		servicesPackage.setDescription("testingServicePackage");
+		servicesPackage = ((PerunBl)perun).getServicesManagerBl().createServicesPackage(perunSession, servicesPackage);
+		
+		// service_service_packages
+		((PerunBl)perun).getServicesManagerBl().addServiceToServicesPackage(perunSession, servicesPackage, testService1);
+		
+		// tasks
+		task = new Task();
+		task.setFacility(facility1);
+		task.setService(testService1);
+		task.setSchedule(0L);
+		task.setStatus(TaskStatus.DONE);
+		List<Destination> destinationsList = new ArrayList<>();
+		destinationsList.add(destination);
+		task.setDestinations(destinationsList);
+		((PerunBl)perun).getTasksManagerBl().insertTask(task);
+		
+		// authz
+		// authz entries for service are removed in ServicesManagerImpl::deleteService(), 
+		// no point in testing here
 	}
 
+	@Test
+	public void testDeleteService() throws Exception {
```

Comment:
You should add a test for every new method on the business layer you added in this PR.

Reply:
Added tests for new methods of TasksManager.

Reply:
Still can't find tests for method listAllTasksForService and probably some others.